lib/icinga/checkable-check.cpp

```diff
@@ -540,6 +490,44 @@ Checkable::ProcessingResult Checkable::ProcessCheckResult(const CheckResult::Ptr
 	if ((stateChange || hardChange) && !children.empty() && (affectsPreviousStateChildren || AffectsChildren()))
 		OnReachabilityChanged(this, cr, children, origin);
 
+	olock->Unlock();
```

Comment:
I rebased the PR yesterday locally before even building it, and I somehow changed this to `olock.Unlock();` :) and was able to build Icinga 2. But the GHAs are now all failing because of this, so please change it as the `operator->` can only be used with pointers. Apart from that, please also add your username into the AUTHORS file.